test-suite/handwritten-src/python/test_containers_interface.py

```diff
@@ -0,0 +1,212 @@
+    assert ci.get_optional_map_string_int() is None
+
+    # Passing non-Nones for the optional arguments
+    set_get(ci, list_int_2, list_int, list_binary, list_binary, list_list_string, \
+     list_record, map_string_int, map_string_int, map_string_string, map_string_string, map_int_list_date, \
+       set_string, set_string, set_string, map_int_set_string, map_optional_int_set_string)
+
+    set_get(ci, list(), list(), list(), list(), list(list()), list(), dict(), dict(), dict(), dict(), dict(), set(), set(), set(), dict(), dict())
+
+    # Passing Nones for the optional arguments
+    set_get(ci, optional_list_int, list_int, list_binary, list_optional_binary, list_list_string, \
+     list_record, optional_map_string_int, map_string_int, map_string_string, map_optional_string_optional_string, map_int_list_date, \
+      optional_set_string, set_string, set_optional_string, map_int_set_string, map_optional_int_set_string)
+
+    # Passing non-None for optional record argument
+    set_get_optional(ci, list(), list(), list(), list(), list(list()), list(), dict(), dict(), dict(), dict(), dict(), set(), set(), set(), dict(), dict())
+    set_get_optional(ci, optional_list_int, list_int, list_binary, list_optional_binary, list_list_string, \
+     list_record, optional_map_string_int, map_string_int, map_string_string, map_optional_string_optional_string, map_int_list_date, \
+      optional_set_string, set_string, set_optional_string, map_int_set_string, map_optional_int_set_string)
+
+    # Passing None for optional record argument
+    ci.set_optional_containers_record(None)
+    get_rec = ci.get_optional_containers_record()
+    assert get_rec is None
+
+    print ("Forcing garbage collection")
+    set_rec = None
+    get_rec = None
+    some_record_2 = None
+    some_record_1 = None
+    list_record = None
+    # set_record = None
+    list_string1 = None
+    list_string2 = None
+    list_list_string = None
+
+    list_date1 = None
+    list_date2 = None
+    map_int_list_date = None
+    gc.collect()
+
+    FooContainersRecord.check_c_data_set_empty()
+    FooSomeOtherRecord.check_c_data_set_empty()
+
+    # print ("GOT MAP ", get_rec._map_string_int)
+    # TODO: more tests: ex: map_binary_list_record
+
+    # print ("got ", get_rec)
+    # print ("_list_list_string", get_rec._list_list_string)
+    # print ("_list_optional_binary", get_rec._list_optional_binary)
+    # print ("_map_opt_str_opt_str", set_rec._map_optional_string_optional_string, get_rec._map_optional_string_optional_string, map_optional_string_optional_string)
+    # print ("_list_record", get_rec._list_record)
+
+def test_set_get_list():
+    ci = FooContainersInterface.create()
+
+    list_binary = list([b'binary_test', b''])
+
+    ci.set_list_binary(list_binary)
+    assert ci.get_list_binary() == list_binary
+    # print ("_list_binary", ci.get_list_binary())
+
+'''
+def test_set_get_list():
+    # TODO: good closer to complete test, for when time allows
```

Comment:
Pls make sure to open an issue for this once we merged into master so we don't forget about it!